gossipsub.go

```diff
@@ -515,19 +519,23 @@ func (gs *GossipSubRouter) sendGraftPrune(tograft, toprune map[peer.ID][]string)
 
 }
 
-func (gs *GossipSubRouter) emitGossip(topic string, peers map[peer.ID]struct{}) {
+// emitGossip emits IHAVE gossip advertising items in the message cache window
+// of this topic.
+func (gs *GossipSubRouter) emitGossip(topic string, exclude map[peer.ID]struct{}) {
 	mids := gs.mcache.GetGossipIDs(topic)
 	if len(mids) == 0 {
 		return
 	}
 
-	gpeers := gs.getPeers(topic, GossipSubD, func(peer.ID) bool { return true })
+	// Send gossip to D peers, skipping over the exclude set.
+	gpeers := gs.getPeers(topic, GossipSubD, func(p peer.ID) bool {
```

Comment:
this is a subtle change as it will send slightly more gossip, but I am ok with it.

Reply:
👍